src/Diagnostics.ModelsAndUtils/Attributes/ResourceFilter/HostingEnvironmentFilter.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Attributes
+{
+    /// <summary>
+    /// Class representing Filter for Hosting Environment Resource
+    /// </summary>
+    public class HostingEnvironmentFilter : ResourceFilterBase
+    {
+        /// <summary>
+        /// Platform Type (Windows, Linux ...)
+        /// </summary>
+        public PlatformType PlatformType;
+
+        /// <summary>
+        /// Hosting Environment Type (ASE V1, ASE V2 ...)
+        /// </summary>
+        public HostingEnvironmentType HostingEnvironmentType;
```

Comment:
I think you can overload StampType here instead of creating a separate type like the HostingEnvironmentType. In the case that someone tries to set this to something that doesn't match ASEv1 or ASEv2 or All we can throw an InvalidOperationException.